ethcore/sync/src/blocks.rs

```diff
@@ -210,11 +210,11 @@ impl BlockCollection {
 	}
 
 	/// Returns a set of block hashes that require a body download. The returned set is marked as being downloaded.
-	pub fn needed_bodies(&mut self, count: usize, _ignore_downloading: bool) -> Vec<H256> {
+	pub fn needed_bodies(&mut self, count: usize) -> Vec<H256> {
 		if self.head.is_none() {
 			return Vec::new();
 		}
-		let mut needed_bodies: Vec<H256> = Vec::new();
+		let mut needed_bodies: Vec<H256> = Vec::with_capacity(count);
 		let mut head = self.head;
 		while head.is_some() && needed_bodies.len() < count {
 			head = self.parents.get(&head.unwrap()).cloned();
```

Comment:
replace this with expect? 

`head is some checked above; qed`?

Reply:
or with  `while let Some(head) = head`

Reply:
I don't think that works because `head` is mutated in the loop?!

Reply:
alternatively
```rust
while needed_bodies.len() < count {
	let head = match head {
		Some(head) => self.parents.get(head).copied(),
		None => break,
	};
```